src/components/InventoryV2/Search.tsx

```diff
@@ -1,38 +1,85 @@
-import { getNurseryName } from './FilterUtils';
 import { useOrganization } from 'src/providers/hooks';
-import { PillList } from '@terraware/web-components';
-import theme from '../../theme';
+import theme from 'src/theme';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { selectSpecies } from 'src/redux/features/species/speciesSelectors';
+import { requestSpecies } from 'src/redux/features/species/speciesThunks';
+import { getNurseryName } from './FilterUtils';
+import InventoryFilters, { InventoryFiltersType } from './InventoryFiltersPopover';
+import { OriginPage } from './InventoryBatch';
 
 interface SearchProps {
   searchValue: string;
   onSearch: (value: string) => void;
   filters: InventoryFiltersType;
   setFilters: React.Dispatch<React.SetStateAction<InventoryFiltersType>>;
+  origin?: OriginPage;
 }
 
-export default function Search(props: SearchProps): JSX.Element {
-  const { selectedOrganization } = useOrganization();
+type PillListItemWithEmptyValue = PillListItem<string> & { emptyValue: unknown };
+
+export default function Search(props: SearchProps): JSX.Element | null {
   const { searchValue, onSearch, filters, setFilters } = props;
-  const [filterPillData, setFilterPillData] = useState<PillListItem<string>[]>();
+  const origin = props.origin || 'Species';
+
+  const { selectedOrganization } = useOrganization();
+  const dispatch = useAppDispatch();
+
+  const species = useAppSelector(selectSpecies);
+
+  const [filterPillData, setFilterPillData] = useState<PillListItemWithEmptyValue[]>([]);
+
+  const getSpeciesName = useCallback(
+    (speciesId: number) => (species || []).find((s) => s.id === speciesId)?.commonName,
+    [species]
+  );
 
   useEffect(() => {
-    let data: PillListItem<string>[] = [];
+    let data: PillListItemWithEmptyValue[] = [];
     if ((filters.facilityIds?.length ?? 0) > 0) {
       data = [
         {
-          id: 'filterPillData',
+          id: 'facilityIds',
           label: strings.NURSERY,
-          value: filters.facilityIds?.map((id) => getNurseryName(id, selectedOrganization))?.join(', ') ?? '',
+          value: filters.facilityIds?.map((id) => getNurseryName(id, selectedOrganization)).join(', ') ?? '',
+          emptyValue: [],
+        },
+      ];
+    } else if ((filters.speciesIds?.length ?? 0) > 0) {
+      data = [
+        {
+          id: 'speciesIds',
+          label: strings.SPECIES,
+          value: filters.speciesIds?.map(getSpeciesName).join(', ') ?? '',
+          emptyValue: [],
         },
       ];
     }
+
     setFilterPillData(data);
-  }, [selectedOrganization, filters.facilityIds]);
+  }, [selectedOrganization, filters.facilityIds, filters.speciesIds, getSpeciesName]);
+
+  useEffect(() => {
+    if (origin === 'Nursery') {
+      void dispatch(requestSpecies(selectedOrganization.id));
+    }
+  }, [origin, dispatch, selectedOrganization.id]);
+
+  const onRemovePillList = useCallback(
```

Comment:
Previously this was [wiping all filters](https://github.com/terraware/terraware-web/pull/1792/files?diff=unified&w=0#diff-2eae7c37411b0359763e941d4410e20f8198e0b617f3b6bf952c399bfab3068cL61), instead of the specific filter that is being removed from the pill list.